src/server/services/post.service.ts

```diff
@@ -64,46 +83,71 @@ export const getPostsInfinite = async ({
   modelVersionId,
   ids,
   collectionId,
+  include,
 }: PostsQueryInput & { user?: SessionUser }) => {
-  const AND: Prisma.Enumerable<Prisma.PostWhereInput> = [];
-  const orderBy: Prisma.Enumerable<Prisma.PostOrderByWithRelationInput> = [];
+  // const AND: Prisma.Enumerable<Prisma.PostWhereInput> = [];
+  const AND = [Prisma.sql`1 = 1`];
+
   const isOwnerRequest = user && user.username === username;
   if (username) {
     const targetUser = await dbRead.user.findFirst({ where: { username }, select: { id: true } });
-    AND.push({ userId: targetUser?.id ?? 0 });
+    if (targetUser) AND.push(Prisma.sql`p."userId" = ${targetUser.id}`);
   }
-  if (modelVersionId) AND.push({ modelVersionId });
+  if (modelVersionId) AND.push(Prisma.sql`p."modelVersionId" = ${modelVersionId}`);
   if (!isOwnerRequest) {
-    AND.push({
-      publishedAt:
-        period === 'AllTime' || periodMode === 'stats'
-          ? { lt: new Date() }
-          : { gte: decreaseDate(new Date(), 1, period.toLowerCase() as ManipulateType) },
-    });
-    if (query) AND.push({ title: { in: query } });
+    AND.push(Prisma.sql`p."publishedAt" < now()`);
+    if (period !== 'AllTime' && periodMode !== 'stats') {
+      AND.push(Prisma.raw(`p."publishedAt" > now() - INTERVAL '1 ${period.toLowerCase()}'`));
+    }
+    if (query) AND.push(Prisma.sql`p.title ILIKE ${query + '%'}`);
     if (!!excludedTagIds?.length) {
-      AND.push({
-        OR: [
-          { userId: user?.id },
-          {
-            tags: { none: { tagId: { in: excludedTagIds } } },
-            helper: { scanned: true },
-          },
-        ],
-      });
+      const excludedOr = [
+        Prisma.sql`NOT EXISTS (
+          SELECT 1 FROM
+          "TagsOnPost" top
+          WHERE top."postId" = p.id
+          AND top."tagId" IN (${Prisma.join(excludedTagIds)})
+        ) AND EXISTS (
+          SELECT 1 FROM "Image" i
+          WHERE i."postId" = p.id AND i."ingestion" = 'Scanned'
+        )`,
+      ];
+      if (user?.id) excludedOr.push(Prisma.sql`p."userId" = ${user.id}`);
+      AND.push(Prisma.sql`(${Prisma.join(excludedOr, ' OR ')})`);
+    }
+    if (!!tags?.length)
+      AND.push(Prisma.sql`EXISTS (
+        SELECT 1 FROM "TagsOnPost" top
+        WHERE top."postId" = p.id AND top."tagId" IN (${Prisma.join(tags)})
+      )`);
+
+    if (!!excludedUserIds?.length)
+      AND.push(Prisma.sql`p."userId" NOT IN (${Prisma.join(excludedUserIds)})`);
+    if (!user?.isModerator) {
+      // Handle Post Visibility
+      AND.push(Prisma.sql`p."modelVersionId" IS NULL OR EXISTS (
+        SELECT 1 FROM "ModelVersion" mv
+        WHERE mv.id = p."modelVersionId" AND mv.status = 'Published'
+      )`);
+
+      // Handle Collection Visibility
+      const collectionOr = [
+        Prisma.sql`p."collectionId" IS NULL`,
+        Prisma.sql`EXISTS (
+        SELECT 1 FROM "Collection" c
+        WHERE c.id = p."collectionId" AND c.read != 'Private'
```

Comment:
I believe we should check if `c.read = 'Public'` since we can have unlisted collections as well. Also, not sure if we need to cast to the enum here, might as well try
```suggestion
        WHERE c.id = p."collectionId" AND c.read != 'Public'::CollectionReadConfiguration
```